### PR TITLE
@hookform/error-message

### PR DESCRIPTION
**Pull request by kotarella1110**
remove `ErrorMessage` component from core lib in v6 (https://github.com/react-hook-form/react-hook-form/pull/1471).
move back to its this repo.

**Reason:** TS support is getting better with  [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) and we could trim some size off our lib size without impacting the end-user experience of rendering errors

See this issue (https://github.com/react-hook-form/react-hook-form/issues/1840) for more detail.

 ```diff
- import { ErrorMessage } from 'react-hook-form';
+ import { ErrorMessage } from '@hookform/error-message'

- <ErrorMessage errors={errors} name="error">
-   {({ message }) => <p>{message}</p>}
- </ErrorMessage>
+ <ErrorMessage
+   errors={errors}
+   name="error"
+   render={({ message }) => <p>{message}</p>}
+ />
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,25 +1,50 @@
...
+    "lint:types": "tsc --noEmit",
+    "test": "jest",
+    "test:watch": "jest --watch",
+    "test:coverage": "jest --coverage"
   },
+  "keywords": [
+    "react",
+    "hooks",
+    "react-hooks"
+  ],
```
Comment (kotarella1110) [inline package.json:33]: @bluebill1049 What are the keywords here?

File: src/types.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import {
+  FieldErrors,
+  FieldName,
+  Message,
+  MultipleFieldErrors,
+  AsProps,
+  Assign,
+} from 'react-hook-form';
```
Comment (kotarella1110) [inline src/types.ts:8]: @bluebill1049 Should I fix core lib to export types?

File: package.json (hunk 1 above)
Comment (bluebill1049) [reply package.json:33]: for SEO :)

File: src/types.ts (hunk 2 above)
Comment (bluebill1049) [reply src/types.ts:8]: i think we can duplicate those types here if they are not essential for users. what do you think?

Comment (bluebill1049) [reply src/types.ts:8]: this component probably not going to be touched after release for a while.

File: package.json [hunk 3]
```diff
@@ -1,55 +1,103 @@
...
+    "react",
+    "react-component",
+    "form",
+    "forms",
+    "form-validation",
+    "validation",
+    "hookform",
+    "react-hook-form",
+    "typescript"
+  ],
```
Comment (kotarella1110) [inline package.json:39]: @bluebill1049 What do you think this keywords?

File: rollup.config.js [hunk 4]
```diff
@@ -1,29 +1,61 @@
...
-      format: 'cjs',
-    },
-    {
-      file: `dist/index.es.js`,
-      format: 'esm',
+export default [
+  {
+    input: 'src/index.tsx',
+    output: {
+      name: 'HookFormErrorMessage',
```
Comment (kotarella1110) [inline rollup.config.js:14]: @bluebill1049 What do you think this umd global naming?
`HookFormErrorMessage` vs `ErrorMessage`

File: package.json (hunk 3 above)
Comment (bluebill1049) [reply package.json:39]: looks great.

File: rollup.config.js (hunk 4 above)
Comment (bluebill1049) [reply rollup.config.js]: `ReactHookFormErrorMessage` ? keep consistent?

Comment (kotarella1110) [reply rollup.config.js]: done!

**Comment by kotarella1110** [comment]
yea! I will publish it! 🚀 
is this version `0.0.1` OK?

**Comment by bluebill1049** [comment]
sounds good @kotarella1110 🚀

**Comment by kotarella1110** [comment]
🎉 This PR is included in version 0.0.1 🎉
https://www.npmjs.com/package/@hookform/error-message

**Comment by bluebill1049** [comment]
awesome!

File: src/ErrorMessage.tsx [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+  ...rest
+}: Props<TFieldErrors, TAs>) => {
+  const methods = useFormContext();
+  const error = get(errors || methods.errors, name);
+
+  if (!error) {
+    return null;
+  }
+
+  const { message: messageFromRegister, types } = error;
```
Comment (qmg-aafridi) [inline src/ErrorMessage.tsx:27]: types don't exist anymore. messages is now always undefined

**Pull request merged**